### PR TITLE
Avoid the deletion of existing PR comments

### PR DESCRIPTION
**Pull request by attiasas**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
---

Introduce a new parameter that offers control over whether to retain previous Frogbot comments from earlier scans on the same PR or, by default, to maintain the existing behavior of deleting them before incorporating the updated information from the current scan.

### CODE REVIEW

File: docs/install-azure-pipelines.md [hunk 1]
```diff
@@ -288,6 +288,10 @@ jobs:
             # [Optional, default: "FALSE"]
             # Displays all existing vulnerabilities, including the ones that were added by the pull request.
             # JF_INCLUDE_ALL_VULNERABILITIES: "TRUE"
+
+            # [Optional, default: "FALSE"]
+            # Keep old comments that were added by previous Frogbot scans on the pull request.
+            # JF_AVOID_OLD_PR_COMMENTS_DELETION: "TRUE"
```
Comment (eyalbe4) [inline docs/install-azure-pipelines.md:294]: JF_AVOID_OLD_PR_COMMENTS_DELETION --> JF_AVOID_PREVIOUS_PR_COMMENTS_DELETION

File: docs/install-azure-pipelines.md [hunk 2]
```diff
@@ -288,6 +288,10 @@ jobs:
             # [Optional, default: "FALSE"]
             # Displays all existing vulnerabilities, including the ones that were added by the pull request.
             # JF_INCLUDE_ALL_VULNERABILITIES: "TRUE"
+
+            # [Optional, default: "FALSE"]
+            # Keep old comments that were added by previous Frogbot scans on the pull request.
```
Comment (eyalbe4) [inline docs/install-azure-pipelines.md:293]: Keep old comments that were added by previous Frogbot scans on the pull request.
-->
When adding new comments on pull requests, keep old comments that were added by previous scans.

File: docs/templates/jfrog-pipelines/pipelines-npm.yml [hunk 3]
```diff
@@ -140,18 +144,6 @@ pipelines:
...
-            # Xray Watches. Learn more about them here: https://www.jfrog.com/confluence/display/JFROG/Configuring+Xray+Watches
-            # JF_WATCHES: <watch-1>,<watch-2>...<watch-n>
-
-            # [Optional, default: "FALSE"]
-            # Displays all existing vulnerabilities, including the ones that were added by the pull request.
-            # JF_INCLUDE_ALL_VULNERABILITIES: "TRUE"
-
-            # [Optional, default: "TRUE"]
-            # Fails the Frogbot task if any security issue is found.
-            # JF_FAIL: "FALSE"
```
Comment (eyalbe4) [inline docs/templates/jfrog-pipelines/pipelines-npm.yml:153]: Were these variable removed intentionally? If so, why?

File: schema/frogbot-schema.json [hunk 4]
```diff
@@ -99,6 +99,11 @@
         "description": "Set to true to display all existing vulnerabilities, including the ones that were not added by the pull request.",
         "title": "Include All Vulnerabilities"
       },
+      "keepPreviousComments": {
+        "type": "boolean",
+        "description": "Keep old comments that were added by previous Frogbot scans on the pull request.",
+        "title": "Keep Previous Frogbot Comments"
+      },
```
Comment (eyalbe4) [inline schema/frogbot-schema.json:106]: Let's uuse the variable name as in the workflows (JF_AVOID_PREVIOUS_PR_COMMENTS_DELETION --> AvoidPreviousPrCommentsDeletion).
Let's also use the same description.

File: utils/consts.go [hunk 5]
```diff
@@ -47,6 +47,7 @@ const (
 	jfrogWatchesEnv              = "JF_WATCHES"
 	jfrogProjectEnv              = "JF_PROJECT"
 	IncludeAllVulnerabilitiesEnv = "JF_INCLUDE_ALL_VULNERABILITIES"
+	KeepPreviousCommentsEnv      = "JF_AVOID_OLD_PR_COMMENTS_DELETION"
```
Comment (eyalbe4) [inline utils/consts.go:50]: Again, consustency is important - 

KeepPreviousCommentsEnv --> AvoidPreviousPrCommentsDeletionEnv

The same guideline goes for the code in `params.go`.

File: utils/reviewcomment.go [hunk 6]
```diff
@@ -26,15 +26,74 @@ const (
 	SastComment       ReviewCommentType = "Sast"
 )
 
-func AddReviewComments(repo *Repository, pullRequestID int, client vcsclient.VcsClient, issues *IssuesCollection) (err error) {
-	if err = deleteOldReviewComments(repo, pullRequestID, client); err != nil {
-		err = errors.New("couldn't delete pull request review comment: " + err.Error())
+func HandlePullRequestCommentsAfterScan(issues *IssuesCollection, repo *Repository, client vcsclient.VcsClient, pullRequestID int) (err error) {
+	if !repo.Params.KeepPreviousComments {
+		log.Debug("Looking for an existing Frogbot pull request comments. Deleting them if exists...")
```
Comment (eyalbe4) [inline utils/reviewcomment.go:31]: ...request comments. Deleting them if exists... --> ...request comment. Deleting it if it exists...

File: utils/reviewcomment.go [hunk 7]
```diff
@@ -26,15 +26,74 @@ const (
...
+		if err = DeleteExistingPullRequestComments(repo, client); err != nil {
+			err = errors.New("couldn't delete pull request comment: " + err.Error())
+			return
+		}
+		if err = deleteExistingPullRequestReviewComments(repo, pullRequestID, client); err != nil {
+			err = errors.New("couldn't delete pull request review comment: " + err.Error())
+			return
+		}
```
Comment (eyalbe4) [inline utils/reviewcomment.go:39]: The above code section includes two deletion function calls - 
1. DeleteExistingPullRequestComments
2. deleteExistingPullRequestReviewComments

It is unclear for the reader what is the diferemce between these two calls. Let's clerify this using a comment here. Let's also add dewcription comments on thsese two functions.

File: utils/reviewcomment.go [hunk 8]
```diff
@@ -53,8 +112,8 @@ func AddReviewComments(repo *Repository, pullRequestID int, client vcsclient.Vcs
 	return
 }
 
-func deleteOldReviewComments(repo *Repository, pullRequestID int, client vcsclient.VcsClient) (err error) {
-	// Get all comments in PR
+func deleteExistingPullRequestReviewComments(repo *Repository, pullRequestID int, client vcsclient.VcsClient) (err error) {
```
Comment (eyalbe4) [inline utils/reviewcomment.go:115]: Let's add a unit test for this function.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: docs/templates/jfrog-pipelines/pipelines-npm.yml (hunk 3 above)
Comment (attiasas) [reply docs/templates/jfrog-pipelines/pipelines-npm.yml:153]: duplicates, appeared above a few rows above
image

File: utils/reviewcomment.go (hunk 8 above)
Comment (attiasas) [reply utils/reviewcomment.go]: done

**Pull request merged**